torch/csrc/autograd/engine.cpp

```diff
@@ -706,10 +736,28 @@ auto Engine::execute(const edge_list& roots,
   // Lock post_callbacks_lock_ before clearing final_callbacks_
   ClearCallbacks _cb_guard(final_callbacks_, post_callbacks_lock_);
 
+  bool is_reentrant_call = worker_device != NO_DEVICE;
+  std::shared_ptr<ReadyQueue> memorized_cpu_ready_queue = nullptr;
+
+  if (is_reentrant_call) {
+    // Reentrant call will still use the parent thread ready_queue
+    // While we can create separate cpu_ready_queue for each new reentrant
+    // thread, but sharing the same cpu_ready_queue with parent thread is
+    // a mild performance improvement and cuda thread still have to do the
+    // same thing.
+    memorized_cpu_ready_queue = local_ready_queue;
+  } else {
+    // not a reentrant call, Engine::execute should start on CPU device
+    // create the thread local ready queue on CPU and memorize it in GraphTask
+    local_ready_queue = std::make_shared<ReadyQueue>();
```

Comment:
I guess this implies that `thread_init` is never called for cPU threads in https://github.com/pytorch/pytorch/pull/33157/files#r378430745  ??

Reply:
Yes, I will guard that local_ready_queue creation under Non-CPU